lib/matplotlib/patches.py

```diff
@@ -3204,23 +3204,110 @@ class _Curve(_Base):
         returned path is the concatenation of the original path, and at
-        most two paths representing the arrow head at the begin point and the
-        at the end point. The arrow heads can be either open or closed.
+        most two paths representing the arrow head or bracket at the begin
+        point and at the end point. The arrow heads can be either open
+        or closed.
         """
 
-        beginarrow = endarrow = False  # Whether arrows are drawn.
+        beginarrow = endarrow = None  # Whether arrows are drawn.
+        arrow = "-"
         fillbegin = fillend = False  # Whether arrows are filled.
 
-        def __init__(self, head_length=.4, head_width=.2):
+        def __init__(self, head_length=.4, head_width=.2, widthA=1., widthB=1.,
+                     lengthA=0.2, lengthB=0.2, angleA=0, angleB=0, scaleA=None,
+                     scaleB=None):
             """
             Parameters
             ----------
             head_length : float, default: 0.4
                 Length of the arrow head, relative to *mutation_scale*.
             head_width : float, default: 0.2
                 Width of the arrow head, relative to *mutation_scale*.
+            widthA : float, default: 1.0
+                Width of the bracket at the beginning of the arrow
+            widthB : float, default: 1.0
+                Width of the bracket at the end of the arrow
+            lengthA : float, default: 0.2
+                Length of the bracket at the beginning of the arrow
+            lengthB : float, default: 0.2
+                Length of the bracket at the end of the arrow
+            angleA : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            angleB : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            scaleA : float, default *mutation_size*
+                The mutation_size for the beginning bracket
+            scaleB : float, default *mutation_size*
+                The mutation_size for the end bracket
             """
+
             self.head_length, self.head_width = head_length, head_width
+            self.widthA, self.widthB = widthA, widthB
+            self.lengthA, self.lengthB = lengthA, lengthB
+            self.angleA, self.angleB = angleA, angleB
+            self.scaleA, self.scaleB = scaleA, scaleB
+
+            self._beginarrow_head = False
+            self._beginarrow_bracket = False
+            self._endarrow_head = False
+            self._endarrow_bracket = False
+
+            _split_arrow = self.arrow.split("-", 1)
+
+            if len(_split_arrow) != 2:
+                raise ValueError("arrow must have the '-' between "
+                                 "the two heads")
```

Comment:
Let's turn this around, and use explicit variables instead of `_split_arrrow[i]` afterwards
```suggestion
            if '-' not in self.arrow:
                raise ValueError("arrow must have the '-' between "
                                 "the two heads")

            beginarrow, endarrow = self.arrow.split("-", 1)
```